### PR TITLE
[JENKINS-61452] Tolerate corrupt Base64 in `PlainTextConsoleOutputStream`

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-61452](https://issues.jenkins.io/browse/JENKINS-61452). Somewhere along the line `ConsoleAnnotationOutputStream` got fixed (perhaps because `java.util.Base64` switched from `IllegalArgumentException` to `IOException`, which was already caught?), but `PlainTextConsoleOutputStream` remained vulnerable to corrupted `ConsoleNote`s.

### Testing done

Unit tests sufficed to reproduce the error.

### Proposed changelog entries

- The plain text console log will still be printed even if some console annotations are corrupt.

### Proposed upgrade guidelines

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8378"></a>

### CODE REVIEW

File: core/src/main/java/hudson/console/PlainTextConsoleOutputStream.java
Comment (dwnusbaum) [inline core/src/main/java/hudson/console/PlainTextConsoleOutputStream.java:76]: Maybe better to just log the raw byte array for debugging purposes since we don't know what charset was used at this level.

**Review by dwnusbaum — APPROVED** [review]

File: core/src/main/java/hudson/console/PlainTextConsoleOutputStream.java
Comment (jglick) [reply core/src/main/java/hudson/console/PlainTextConsoleOutputStream.java:76]: Well, normally ASCII for the note itself, but the `escapeJava` will take care of any escapes. Note that Pipeline is always UTF-8, and Java will always be UTF-8 in a future version, so I am not too concerned.

BTW this is simply copied from https://github.com/jenkinsci/jenkins/blob/83085bac0081f7366abf9c74616b1b1a9ad62bb3/core/src/main/java/hudson/console/ConsoleAnnotationOutputStream.java#L125

**Review by Vlatombe — APPROVED** [review]
LGTM AFAIU

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
